01/frontend/src/components/App.tsx

```diff
@@ -1,15 +1,44 @@
-import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
-import Form from "./Form";
-import LoggedIn from "./LoggedIn";
+import { useEffect } from "react";
+import { Route, Routes } from "react-router-dom";
+import Form from "@components/Form";
+import LoggedIn from "@components/LoggedIn";
+import useAuth from "@hooks/useAuth";
+import React from "react";
+import RedirectIfNotAuthenticated from "./RedirectIfNotAuthenticated";
 
 const App = () => {
+  const { authenticate } = useAuth();
+
+  useEffect(() => {
+    if (localStorage.getItem("jwtToken") !== null) {
+      authenticate();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const publicRoutes = [{ path: "/", component: <Form /> }];
```

Comment:
What is the Form component?. In my opinion name of this component is too generic.